app/scripts/modules/core/account/account.service.ts

```diff
@@ -0,0 +1,210 @@
+import {chain, intersection, memoize} from 'lodash';
+import {module} from 'angular';
+
+import {Application} from 'core/application/application.model';
+import {API_SERVICE, Api} from 'core/api/api.service';
+
+export interface IRegion {
+  name: string;
+  preferredZones: string[];
+  availablityZones: string[];
+}
+
+export interface IAccount {
+  accountId: string;
+  name: string;
+  regions: IRegion[];
+  requiredGroupMembership: string[];
+  type: string;
+}
+
+export interface IZone {
+  [key: string]: string[];
+}
+
+export interface IAccountZone {
+  [key: string]: IZone;
+}
+
+export class AccountService {
+
+  static get $inject() {
+    return ['$log', '$q', 'settings', 'infrastructureCaches', 'cloudProviderRegistry', 'API'];
+  }
+
+  constructor(private $log: ng.ILogService,
+              private $q: ng.IQService,
+              private settings: any,
+              private infrastructureCaches: any,
+              private cloudProviderRegistry: any,
+              private API: Api) {
+  }
+
+  public challengeDestructiveActions(account: string): ng.IPromise<any> {
+    return memoize(() => {
+      return this.$q((resolve: ng.IQResolveReject<any>) => {
+        if (account) {
+          this.getAccountDetails(account)
+            .then((details: any) => {
+              if (details) {
+                resolve(details.challengeDestructiveActions);
+              } else {
+                resolve(false);
+              }
+            })
+            .catch(() => resolve(false));
+        } else {
+          resolve(false);
+        }
+      });
+    })();
+  }
+
+  public getAccountDetails(account: string): any {
+    return this.API.one('credentials')
+      .one(account)
+      .useCache()
+      .get();
+  }
+
+  public getAllAccountDetailsForProvider(provider: string): ng.IPromise<any> {
+    return memoize(() => {
+      return this.listAccounts(provider)
+        .then((accounts: IAccount[]) => this.$q.all(accounts.map((account: IAccount) => this.getAccountDetails(account.name))))
+        .catch((error: any) => {
+          this.$log.warn(`Failed to load accounts for provider "${provider}"; exception:`, error);
+          return [];
+        });
+    })();
+  }
+
+  public getAvailabilityZonesForAccountAndRegion(provider: string, account: string, region: string): ng.IPromise<any[]> {
```

Comment:
would be nice to get `<any[]>` as `<string[]>`, although that would probably mean reworking the interface for `IAccountZone`